src/calibrate-multi-hop/models/calibrate.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import pandas as pd
+import os
+from pymongo import MongoClient
+from simple_calibration import simple as sp 
+from datetime import datetime,timedelta
+
+
+
+class Calibrate():
+    """
+        The class contains functionality for computing device calibrated values .
+    """
+    def __init__(self):
+        """ initialize """
+
+        with open('models/encounters.p','rb') as pickle_file:
+            encounters = pd.read_pickle(pickle_file)
+            self.allcals, self.myDict_id = self.calibrate_raw_data(encounters)
+
+    def calibrate_raw_data(self, encounters):
+        unq = np.unique(np.r_[encounters['channel_id_sensorA'].unique(),encounters['channel_id_sensorB'].unique()])
+        t = (encounters['created_at']-pd.Timestamp('2020-07-15',tz='UTC')).dt.total_seconds()/3600 #hours since 15th July put to 1970 and substract some dates
+        idA = [np.where(a==unq)[0][0] for a in encounters['channel_id_sensorA']]
+        idB = [np.where(a==unq)[0][0] for a in encounters['channel_id_sensorB']]
+        sA = np.nanmean(encounters[['pm2_5_sensorA','s2_pm2_5_sensorA']],1)
+        sB = np.nanmean(encounters[['pm2_5_sensorB','s2_pm2_5_sensorB']],1)
+        X = np.c_[t,idA,idB]
+        Y = np.c_[sA,sB]
+
+        newids = set(idA+idB)
+        myDict_id = dict(zip(unq, newids))
+
+        refsensor = np.zeros(len(unq))
+        refsensor[1]=1
+        f = sp.f
+        delta = 24*7
+        G,allsp,allcals,allcallists,allpopts,allpcovs,allpoptslists = sp.compute_simple_calibration(X,Y,delta,refsensor)
+        return allcals, myDict_id
+   
+  
+    def calibrate_sensor_raw_data(self, datetime, sensor_id, raw_value):
+        
+        delta = 24*7
+
+        datetime = pd.Timestamp(datetime, tz='UTC')
+        time = (datetime-pd.Timestamp('2020-07-15',tz='UTC')).total_seconds()/3600  
```

Comment:
Hi @bobturneruk 
Could we have a short chat (Hangouts call) early next week to review this implementation? I might appreciate some aspects on a call perhaps.....potential candidates on that suggested **short** working could be @NoahNsimbe and @ja-odur .....thanks.....